src/App.js

```diff
@@ -1,19 +1,38 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
+// import ChatEntry from './components/ChatEntry';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const updateToggleLike = (updatedEntry) => {
+    const newChatData = chatData.map((entry) => {
+      if (entry.id === updatedEntry.id){
+        return updatedEntry;
+      } else {
+        return entry;
+      }
+    }); 
+    setChatData(newChatData)
+  }
+
+  const totalLikes = chatData.reduce((total, entry) => {
+    return total + entry.liked;
+  }, 0);
```

Comment:
Since this logic is not in a method at first glance, it can be a little unclear when this should execute and what it's doing. To help organize your code, put this logic into a method like `getHeartCount` and then you can invoke the method in the JSX.

So line 29 would look like <section>{getHeartCount()} ❤️s</section>